src/com/hms/flexyosisoftconnector/Tag.java

```diff
@@ -120,6 +170,264 @@ public void recordTagValue(String time)
+               int lastInterestingSemiColon = 4;
+               for(int i = 0; i < lastInterestingSemiColon; i++) {
+
+                   // need to add 1 to index main so indexOf function can work
+                   // or it will just find the same thing again
+                   indexMain = 1+ line.indexOf(';', indexMain);
+                   if(i == firstSemiColonOfInterest) {
+                       indexStart = indexMain;
+                   }
+                   if(i == secondSemiColonOfInterest) {
+                       // subtract one here to exclude the semicolon from the substring
+                       indexEnd = indexMain-1;
+                   }
+               }
+               value = line.substring(indexStart, indexEnd);
+
+               // below checks are to only record into internal queue if it is a new tag in historical logs
+
+               // if last data point is not null, check the new one is new
+               if (lastDataPoint == null) {
+                   // there was no last point, start of program condition
+                   recordTagValue(timeInt, value);
+               } else {
+                   // convert last point's epoch time to seconds, which is what the EBD "timeInt" is in
+                   long lastPointSeconds = lastDataPoint.epochTime/1000;
+                   if (lastPointSeconds != Long.parseLong(timeInt)) {
+                       recordTagValue(timeInt, value);
+                   }
+               }
+
+               // read next line, repeat
+               line = reader.readLine();
+           }
+       } catch (IOException e) {
+           Logger.LOG_ERR("IO exception in Tag Class, reading in historical log file.");
+           e.printStackTrace();
+       }
+   }
+
+
+   /**
+    * converts given long into date string of expected format
+    * and writes it to a file
+    *
+    * @param timeInt epoch time in seconds to get converted and stored in file
+    * @return just return what string it set
+    */
+   public String writeTimePointerFile(long timeInt)
+   {
+      // convert from epoch time long to string
+      String fileValueString = convertTimeIntHelper(timeInt);
+
+
+      File file = new File(tagTimeIntFileName);
+      FileWriter fr = null;
+      BufferedWriter br = null;
+      try{
+          fr = new FileWriter(file);
+          br = new BufferedWriter(fr);
+
+          // write the data to the file here
+          br.write(fileValueString);
+
+      } catch (IOException e) {
+          e.printStackTrace();
+      }finally{
+          try {
+              br.close();
+              fr.close();
+          } catch (IOException e) {
+              e.printStackTrace();
+          }
+      }
+      return fileValueString;
+   }
+
+   /**
+    * This function is only called when retrieving value for call to export block descriptor.
+    *
+    * the tagTimeIntFileName file holds a time pointer for every tag. 1 file exists per tag.

```

Comment:
A lot of your comments begin sentences with a lower case but in this case you are switching between the two. I would suggest going through your comments and starting all sentences with capitals.